runtime/runtime/src/actions.rs

```diff
@@ -674,6 +676,38 @@ pub(crate) fn action_deploy_contract(
     Ok(())
 }
 
+/// Compiles the permanent contract and store the compiled module in the cache.
```

Comment:
Is it a common practice to compile contracts on deployment? Is it meant to filter out invalid contracts?